### ISSUE TITLE
"components should block"

### ISSUE THREAD

**Issue by optimismus** [CBrosius/luxtronik_v1#8]
This is inside my ESP logging:
`[20:34:41][D][luxtronik_v1:116]: S: 1200 - 0
[20:34:41][W][component:204]: Component <unknown> took a long time for an operation (0.14 s).
[20:34:41][W][component:205]: Components should block for at most 20-30ms.
[20:34:41][D][luxtronik_v1:212]: 1200 found -> Inputs`

I think the ESP need some time to work with the results?! May be there should be a little pause between the calls?

**Comment by CBrosius** [comment]
Yes this is a warning, that is shown now by ESPHome since 2023.7.0
There is an issue (https://github.com/esphome/issues/issues/4717) in the esphome-git for this already open.

`And here, basically we have two choices:

The sensor requires a nudge to send back the value over the bus:
in update() we write the "update_me" command, we return
in loop() we read the response and publish the requested sensors as we receive them, if the sensor requires more command for the other server, we send the command after each read(). We need a local buffer to cover those responses. And state tracking
The sensor streams value:
in loop(): we read values and update local variables
in update(): we publish the values we have available.
Unfortunately this won't solve the case of eInk displays. There is no way current technology would refresh a screen in 30ms. That is an entirely different matter. In general with the stream approach you need to fall back to continuations. Anyone who has used JavaScript before Promise knows where that staircase will eventually lead.`

Maybe we can try to include some update() lines in the loop()-codepart.

**Comment by cbrosius** [comment]
with the actual version I rewrote the parsing logic and got rid of mostly all blocking warnings.
so I will mark this issue as fixed
